### PR TITLE
feat: migration to Alloy

### PR DESCRIPTION
**Pull request by Evalir**

### CODE REVIEW

File: test-data/test-contract-versions/pinned-0.4.14.sol
Comment (mattsse) [inline test-data/test-contract-versions/pinned-0.4.14.sol:1]: what's with these tests?

**Review by mattsse — APPROVED** [review]
cool, this was simple, 

pending q re solidity test data

File: test-data/test-contract-versions/pinned-0.4.14.sol
Comment (Evalir) [reply test-data/test-contract-versions/pinned-0.4.14.sol:1]: Noticed `alloy_json_abi::JsonAbi` doesn't parse pre-`0.5.x` ABIs and we had these test expecting this to be the case from ethabi.  This shouldn't be an issue for foundry as ds-test is `>=0.5.0` though, and we should prob just remove them00.

**Review by gakonst — APPROVED** [review]
let's cut the commented out tests and add a note about that in the readme -- lgtm

**Pull request merged**